doc/2/framework/classes/request/setResult/index.md

```diff
@@ -0,0 +1,54 @@
+---
+code: true
+type: page
+title: setResult
+description: Request class setResult() method
+---
+
+# setResult
+
+Sets the request result and status
+
+### Arguments
+
+```ts
+setResult(
+  result: any,
+  options: {
+    status?: number
+    headers?: JSONObject;
+    raw?: boolean;
+  }): void;
+```
+
+</br>
+
+| Name | Type | Description                      |
+|------|------|----------------------------------|
+| `result` | <pre>any</pre> | Request result. Will be converted to JSON unless `raw` option is set to `true` |
+| `options` | <pre>JSONObject</pre> | Optional parameters |
+
+The `options` argument may contain the following properties:
+
+| Name | Type | Description                      | Default |
+|------|------|----------------------------------|---------|
+| `status` | <pre>integer</pre> | HTTP status code | `200` |
+| `headers` | <pre>JSONObject</pre> | Additional response protocol headers | `null` |
+| `raw` | <pre>boolean</pre> | Instead of a Kuzzle response, forward the result directly | `false` |
+
+If a `KuzzleError` is provided, the request's status attribute is set to the error one.
+
+Otherwise, the provided error is encapsulated into an `InternalError` object, and the request's status is set to 500.
```

Comment:
`Request.setResult ` does not accept error results, and throws if one is provided.

Reply:
I took the documentation from `kuzzle-common-objects` README :sweat_smile:

Reply:
I removed this part :+1: